code/game/objects/items/weapons/cigs.dm

```diff
@@ -19,57 +20,51 @@ LIGHTERS ARE IN LIGHTERS.DM
 	name = "cigarette"
 	desc = "A roll of tobacco and nicotine."
 	icon_state = "cigoff"
-	throw_speed = 0.5
 	item_state = "cigoff"
+	throw_speed = 0.5
 	slot_flags = SLOT_FLAG_MASK
 	w_class = WEIGHT_CLASS_TINY
 	body_parts_covered = null
 	attack_verb = null
 	container_type = INJECTABLE
+	/// Is the cigarette lit?
 	var/lit = FALSE
+	/// Lit cigarette sprite.
 	var/icon_on = "cigon"  //Note - these are in masks.dmi not in cigarette.dmi
+	/// Unlit cigarette sprite.
 	var/icon_off = "cigoff"
+	/// Are we an extra-classy smokable?
+	var/fancy = FALSE
+	/// What trash item the cigarette makes when it burns out.
 	var/type_butt = /obj/item/cigbutt
-	var/lastHolder = null
-	var/smoketime = 150
+	/// How long does the cigarette last before going out? Decrements by 1 every cycle.
+	var/smoketime = 150 // 300 seconds.
+	/// The cigarette's total reagent capacity.
 	var/chem_volume = 60
+	/// A list of the types and amounts of reagents in the cigarette.
 	var/list/list_reagents = list("nicotine" = 40)
-	var/first_puff = TRUE // the first puff is a bit more reagents ingested
+	/// Has anyone taken any reagents from the cigarette? The first tick gives a bigger dose.
+	var/first_puff = TRUE
 	sprite_sheets = list(
 		"Vox" = 'icons/mob/clothing/species/vox/mask.dmi',
 		"Unathi" = 'icons/mob/clothing/species/unathi/mask.dmi',
 		"Tajaran" = 'icons/mob/clothing/species/tajaran/mask.dmi',
 		"Vulpkanin" = 'icons/mob/clothing/species/vulpkanin/mask.dmi',
 		"Grey" = 'icons/mob/clothing/species/grey/mask.dmi')
 
-	var/static/things_that_light = typecacheof(list(
-		/obj/item/lighter,
-		/obj/item/match,
-		/obj/item/melee/energy/sword/saber,
-		/obj/item/assembly/igniter,
-		/obj/item/gun/magic/wand/fireball))
-
-
 /obj/item/clothing/mask/cigarette/Initialize(mapload)
 	. = ..()
 	create_reagents(chem_volume) // making the cigarrete a chemical holder with a maximum volume of 30
 	reagents.set_reacting(FALSE) // so it doesn't react until you light it
 	if(list_reagents)
 		reagents.add_reagent_list(list_reagents)
 	smoketime = reagents.total_volume * 2.5
-	RegisterSignal(src, COMSIG_ITEM_BEING_ATTACKED, PROC_REF(try_light))
 
 /obj/item/clothing/mask/cigarette/Destroy()
 	QDEL_NULL(reagents)
 	STOP_PROCESSING(SSobj, src)
 	return ..()
 
-/obj/item/clothing/mask/cigarette/proc/try_light(obj/item/cigarette, obj/item/lighting_item)
-	SIGNAL_HANDLER
-	if(lighting_item.get_heat())
-		light()
-		return COMPONENT_CANCEL_ATTACK_CHAIN
```

Comment:
Just make sure you don't break the thing that this signal was set up for

Reply:
It all still works, it's just signal-less now.